extensions/ql-vscode/src/databases/local-databases-ui.ts

```diff
@@ -797,12 +817,119 @@ export class DatabaseUI extends DisposableObject {
     progress: ProgressContext | undefined,
   ): Promise<DatabaseItem | undefined> {
     if (this.databaseManager.currentDatabaseItem === undefined) {
-      await this.chooseAndSetDatabase(false, progress);
+      progress?.progress({
```

Comment:
Is it possible to remove one of the `progress` here? Similar to how it's done in https://github.com/github/vscode-codeql/blob/koesie10/code-server-docker-compose/extensions/ql-vscode/src/databases/database-fetcher.ts#L143

Reply:
It's because `progress` here is a `ProgressContext` instead of a regular `ProgressCallback`. I'm not totally sure why this code uses `ProgressContext`, but it doesn't seem to be something added by this PR.

You could do something like change the name of `progress` to `progressContext` (or whatever you want) and then do
```typescript
const { progress, token } = progressContext;
```
and then you can use `progress` and `token` as normal.

Or you could go further and basically remove the `ProgressContext` type. As far as I can tell, we only use it from this file, and we only ever use the `progress` field from it and never the `token` field. So arguably we could clean up this code and change all references to `ProgressContext` to just use `ProgressCallback` instead.

If that didn't make any sense I'm happy to explain more, or pair on it. We could also save it for a separate PR after this one.

Reply:
Thanks! For now, I've just renamed the first `progress` to `progressContext` to avoid some confusion. Agreed that the other changes would be nice, but slightly out of scope for this PR 😅